test/e2e/specs/shared-blocks.test.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * Internal dependencies
+ */
+import '../support/bootstrap';
+import {
+	insertBlock,
+	newDesktopBrowserPage,
+	newPost,
+	pressWithModifier,
+	searchForBlock,
+} from '../support/utils';
+
+function waitForAndAcceptDialog() {
+	return new Promise( ( resolve ) => {
+		page.once( 'dialog', async ( dialog ) => {
+			await dialog.accept();
+			resolve();
+		} );
+	} );
+}
+
+describe( 'Shared Blocks', () => {
+	beforeAll( async () => {
+		await newDesktopBrowserPage();
+		await newPost();
+	} );
+
+	beforeEach( async () => {
+		// Remove all blocks from the post so that we're working with a clean slate
+		await page.evaluate( () => {
+			const blocks = wp.data.select( 'core/editor' ).getBlocks();
+			const uids = blocks.map( ( block ) => block.uid );
+			wp.data.dispatch( 'core/editor' ).removeBlocks( uids );
+		} );
+	} );
+
+	it( 'can be created', async () => {
+		// Insert a paragraph block
+		await insertBlock( 'Paragraph' );
+		await page.keyboard.type( 'Hello there!' );
+
+		// Trigger isTyping = false
+		await page.mouse.move( 200, 300 );
+		await page.mouse.move( 250, 350 );
+
+		// Convert block to a shared block
+		await page.click( 'button[aria-label="More Options"]' );
+		const convertButton = await page.waitForXPath( '//button[text()="Convert to Shared Block"]' );
```

Comment:
Selecting elements looks like fun - have we considered using data attributes on our elements to make writing tests a bit easier? e.g. `data-test-name: "convert-to-shared-block-button"`. We could also make test objects for the components containing the selectors so that tests are easier to write `const convertButton = await page.waitForXPath( convertToSharedBlockButton.root );`

Reply:
made an issue here:
https://github.com/WordPress/gutenberg/issues/7974